### ISSUE TITLE
Complete migration to v2 primitives & parachain-host

### ISSUE THREAD

**Issue by rphmeier** [paritytech/polkadot#4803]
We're currently using a mix of v1 and v2, which is not great. We should move everything to v2 and fully define v2. Re-exporting stuff from v1 is fine when it's still the same as v1.

**Comment by rphmeier** [comment]
This means updating everything to `use polkadot_primitives::v2::...` whenever it just works and adding code to handle the differences where necessary. Might need a runtime upgrade first. Further investigation needed, let's discuss here.

**Comment by eskimor** [comment]
Actually, when reading through your `Element` message, it would make more sense to me if we did not import v1 primitives, but instead just start v2 as a copy of v1 and have modifications then. v1 should then be treated as immutable and we can delete it eventually without any issues.

**Comment by ordian** [comment]
Initially I made `primitives::v2` re-export all of the `primitives::v1` (except for `SessionInfo`). This way we'd switch all code to `primitives::v2`, but keep `v1` immutable. But it turned out to be a pain to constantly resolve all the merge conflicts until it was merged. If we want to bundle a few other changes, I'd suggest introducing `v3` and moving `v1` and `v2` into that.

**Comment by rphmeier** [comment]
> instead just start v2 as a copy of v1 and have modifications then. v1 should then be treated as immutable and we can delete it eventually without any issues.

Yeah, this is the most 'pure' thing to do but it also means that all subsystems need to be totally adapted to the new v2 types.

**Comment by rphmeier** [comment]
For example, I think that the `RuntimeApiRequest` enum should be renamed `RuntimeApiV1Request` and that `RuntimeApiMessage::Request` should be named `RuntmeApiMessage::RequestV1`. Then we can have V1, V2, VStaging (VStaging is the 'workbench' spec which is flexible, whereas others aren't).

**Comment by rphmeier** [comment]
So the specific proposal for work here is 
  1. Explicitly rename `RuntimeApiRequest` to `RuntimeApiV2Request`
  2. Add a `RuntimeApiVersion` request message to `RuntimeApiVersion`
  3. Update `subsystem_util` logic to place helpers for this stuff into a `v2` module
  4. Update all imports of `polkadot-primitives::v1::*` to use `v2`
  5. Remove `polkadot-primitives::v0` and `polkadot-primitives::v1` and move everything necessary into v2
  6. Ensure that `v2`

@ordian I guess this can be done as soon as the v2 API hits Kusama & Polkadot?

**Comment by ordian** [comment]
> @ordian I guess this can be done as soon as the v2 API hits Kusama & Polkadot?

0.9.17 (for Polkadot) has been branched off so we're not blocked on this to be merged into master
kusama is already on v2
